### ISSUE TITLE
undefined metric warning

### ISSUE THREAD

**Issue by Hussain2111** [Hussain2111/StockPrice-Predictor#1]
UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 due to no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, f"{metric.capitalize()} is", len(result)).                                                                                                                           
                                                                                                                                                                                                            ChatGPT response:                                                                                                                                                                                            from sklearn.metrics import precision_score

# Assuming model and test[predictors] are defined
preds = model.predict(test[predictors])
preds = pd.Series(preds, index=test.index)

# Calculate precision score with zero_division parameter set to 0
precision = precision_score(test["Target"], preds, zero_division=0)
print(f"Precision: {precision}")
